### PR TITLE
Fixes #16755 - properly return puppet path when updating

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://github.com/Katello/katello/blob/master/app/models/katello/content_view_puppet_environment.rb#L73  is expecting this method to return a string

File: app/models/katello/concerns/smart_proxy_extensions.rb [hunk 1]
```diff
@@ -58,6 +58,7 @@ def update_puppet_path
           path = ProxyAPI::PulpNode.new(:url => self.url).capsule_puppet_path['puppet_content_dir']
         end
         self.update_attribute(:puppet_path, path) if path
```
Comment (zjhuntin) [inline app/models/katello/concerns/smart_proxy_extensions.rb]: Should we also remove this conditional since it should always have the path?

Comment (jlsherrill) [inline app/models/katello/concerns/smart_proxy_extensions.rb]: updated

**Comment by zjhuntin** [comment]
APJ!

**Pull request merged**